service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/LagomConfigComponent.scala

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.api
+
+import com.typesafe.config.Config
+import play.api.Configuration
+
+trait LagomConfigComponent {
+  def configuration: Configuration
+  def config: Config = configuration.underlying
+}
```

Comment:
What I mean is that we should stop using `configuration` ourselves.

In [`LagomApplicationLoader`](https://github.com/lagom/lagom/pull/881/files#diff-cf12adb26eaa0af0e97ae7f991d242dcR233) we define `configuration` by building it up from various layers. Then, we define `config` as `configuration.underlying`.

But if we're trying to deprecate `configuration`, then we should switch that: build up a `Config` object as the "source of truth", then wrap it in a `play.api.Configuration` only where we need to interface with Play.